test/common/api_utils.py

```diff
@@ -0,0 +1,170 @@
+from on_http_api2_0 import ApiApi as Api
+from config.api2_0_config import config
+from json import loads
+import sys
+import os
+
+
+def api_node_select_argsList(nodeType='compute', validate_obm=False):
+    """
+    This routine produces a list of node Id's that satisfy the requested parameters. 
+    Nodes may be filtered by Node Id,  SKU name, OBM MAC address which are defined
+    in the system environment.  The node type and OBM setting may be specified.
+    :param  nodeType: Node type (i.e. compute) (default= 'compute')
+    :param  validate_obm: should the node have OBM settings (default=False)
+    return: A list with node IDs that match node type, SKU, and possible valid OBM settings.
+    """
+    return api_node_select(config.api_client,
+                           nodeId=os.getenv("NODEID", "None"),
+                           skuName=os.getenv("SKU", "all"),
+                           obmmac=os.getenv("OBMMAC", "all"),
+                           nodeType=nodeType,
+                           validate_obm=validate_obm,
+                           verbosity=os.getenv("VERBOSITY", "0")
+                           )
+
+
+
+def api_node_select(client,
+                    nodeId=None,
+                    skuName='all',
+                    obmmac='all',
+                    nodeType='compute',
+                    validate_obm=False,
+                    verbosity=None
+                    ):
+    """
+    This routine produces a list of node Id's that satisfy the requested parameters. 
+    Nodes may be filtered by Node Id,     SKU name, OBM MAC address, or node type. 
+    The OBM setting may also be validated (validate_obm).
+    Note: Validation of obm setting is not provides if node id is specified.
+    :param  client: reference to config.api2_0_config
+    :param  nodeId: ID of node object (default=None)
+    :param  skuNmae: Name of sku (default='all')
+    :param  obmmac: OBM mac address (default='all')
+    :param  nodeType: Node type (i.e. compute) (default= 'compute')
+    :param  validate_obm: should the node have OBM settings (default=False)
+    return: A list with node IDs that match node type, SKU, and possible valid OBM settings.
+    """
+
+    nodeIdList = []
+    skuid = "None"
+
+    # set verbosity level
+    if not verbosity:
+        verbosity = os.getenv("VERBOSITY", "0")
+
+    # if user specified single node id, don't bother getting sku information
+    if not nodeId:
+        #
+        # Get a list of the current SKU objects
+        #
+        Api().skus_get()
+        skulistrsp = client.last_response
+        skulist = loads(skulistrsp.data)
```

Comment:
Api().sku.get() does not return a object containing status or data. It's another problem with Api().